src/plugins/preauth/pkinit/pkinit_identity.c

```diff
@@ -610,8 +610,7 @@ pkinit_identity_prompt(krb5_context context,
             retval = crypto_cert_select_default(context, plg_cryptoctx,
                                                 req_cryptoctx, id_cryptoctx);
             if (retval) {
-                pkiDebug("%s: Failed while selecting default certificate\n",
-                         __FUNCTION__);
+                TRACE_PKINIT_NO_DEFAULT_CERT(context);
```

Comment:
This can only fail in the OpenSSL implementation because there is more than one cert to choose from.  The trace message in crypto_cert_select_default() seems more informative in that case, and we don't need two messages.

For parallelism we might also want pkinit_cert_matching() to be responsible for tracing when there is no matching cert.